python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/sling/resources.py

```diff
@@ -410,9 +428,10 @@ def replicate(
 
         for stream in stream_definition:
             output_name = dagster_sling_translator.get_asset_key(stream)
-            yield MaterializeResult(
-                asset_key=output_name, metadata={"elapsed_time": end_time - start_time}
-            )
+            if isinstance(context, AssetExecutionContext):
+                yield MaterializeResult(
+                    asset_key=output_name, metadata={"elapsed_time": end_time - start_time}
+                )
```

Comment:
@rexledesma is this an adequate way of materializing a result in the appropriate context?

Reply:
Use `has_asset_def`, since folks could be typing their assets with `OpExecutionContext` still: https://github.com/dagster-io/dagster/blob/ba96a6925396f49c5aa05a10599f403eb27e6455/python_modules/dagster/dagster/_core/execution/context/compute.py#L568